src/js/components/common/NavLink.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Link } from 'react-router';
+import jquery from 'jquery';
+
+class NavLink extends React.Component {
+  generateNavLink() {
+    let links = Object.keys(this.props.subLinks).map((key) => {
+      return (
+        <Link
+          key={key}
+          to={`#${key}`}
+          className="page-scroll dropdown-item"
+          onClick={this.props.navLinkOnClick}
+        >
+          {this.props.subLinks[key]}
+        </Link>
+      )
+    });
+
+    if (window.location.pathname === this.props.path && Object.keys(this.props.subLinks).length) {
+      return (
+        <li className="nav-item dropdown">
+          <Link
+            className="nav-link dropdown-toggle"
+            onlyActiveOnIndex
+            activeClassName="active"
+            to="#"
+            data-href={this.props.path}
+            id="navbarDropdownMenuLink"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+          >
+            {this.props.name} <span className="caret down"></span>
+          </Link>
+          <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
+            {links}
+          </div>
+        </li>
+      )
+    }
+
+    return (
+      <li className="nav-item">
+        <Link
+          to={this.props.path}
+          activeClassName="active"
+          onlyActiveOnIndex
+          className="nav-link"
+        >
+          {this.props.name}
+        </Link>
+      </li>
+    )
```

Comment:
Love the idea of `generateActiveLink()`. We could make it be even more specific and know only what it needs to know by extracting the components themselves to separate classes. 

For e.g., we could have `let links = Object.keys(this.props.subLinks).map(key => <SubLink />)` 

This is nice to keep in mind, not necessarily to effect on this one. Perhaps a technical debt.